src/lib/extension-load-queue.ts

```diff
@@ -0,0 +1,86 @@
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type BrowserWindow = any;
+
+export type ControllerLoadedCallback<T> = (controller: T) => void;
+
+export const QUEUE_SYMBOL: unique symbol = Symbol.for('queue');
+
+export type ScriptStore<T> = ControllerLoadedCallback<T>[] | null;
+
+export interface CreateLoadQueueArgs<T> {
+    store: ScriptStore<T>;
+    createController: () => T;
+    isQueueCreated?: boolean;
+    onQueueCreated?: (created: boolean) => void;
+}
+
+export const isBrowser = () => typeof window !== 'undefined' && typeof document !== 'undefined';
```

Comment:
As I imagine, `isBrowser` should be a very frequently used utility, so if we're at all serious about this package and its usage, i'd place this in a separate top-level file and would make sure to document this.

Reply:
I agree, at the same time, the current version implies use in place - I'll leave it like that for now, or rename it. I'll also add a comment about correcting it in the future.